src/widgetastic/widget.py

```diff
@@ -1774,8 +1782,10 @@ def _filtered_rows(self, *extra_filters, **filters):
         rows = []
         for row_element in self.browser.elements(query, parent=self):
             row_pos = self._get_number_preceeding_rows(row_element)
-            row_pos = row_pos if not self._is_header_in_body else row_pos + 1
-            rows.append(self.Row(self, row_pos, logger=create_item_logger(self.logger, row_pos)))
+            row_pos = row_pos if self._is_header_in_body else row_pos + 1
```

Comment:
*REQUIRED*
I'm confused by this. If there's 4 preceeding rows and `_is_header_in_body == True` then `index` for the row (after `TableRow.__init__` runs) will be 5. If there's 4 preceeding rows and `_is_header_in_body == False` then `index` for the row will also be 5. If this is working right can you add a comment to help people like me out :P

Reply:
This has to do with number_proeceeding_rows using JS to calculate the index and not factoring in the header.

Let's assume our table has 5 rows of data, and a header, we're instantiating the last row (index 5)

For table1 with thead element, `_get_number_preceeding_rows` does not count the thead, and there are 4 preceeding elements returned. We then add 1 since `_is_header_in_body` is false, and TableRow has index=5.

For table2 withOUT thead element, `_get_number_preceeding_rows` does count the header, as its a <tr> element that the javascript gobbles up. So 5 preceeding elements is returned as the count - and we have a correct index in that case. No need to add 1.

I'll put this in-line as comments.